src/averageAges.js

```diff
@@ -55,7 +69,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person =>
+    people.some(mom => (person.name === mom.mother && person.sex === 'f')));
+  const children = people.filter(person =>
+    mothers.some(mom => (person.mother === mom.name)));
+  const onlySon = children.filter(son => son.sex === 'm');
+
+  const allAge = children.map(child =>
+    child.born - mothers.find(mom => child.mother === mom.name).born);
+  const sonsAge = onlySon.map(child =>
+    child.born - mothers.find(mom => child.mother === mom.name).born);
+
+  const averageAllAge = allAge.reduce((sum, age) => sum + age, 0)
+    / allAge.length;
+  const averageSonsAge = sonsAge.reduce((sum, age) => sum + age, 0)
```

Comment:
you repeat this block of code in all functions. Please, create helper function for it